vitrina/smart_contracts/utils.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+import hashlib
+import xml.etree.ElementTree as ET
+import zipfile
+
+from pdfminer.high_level import extract_text
+
+from vitrina.smart_contracts.exceptions import InvalidAdocError
+
+MANIFEST_FILE_PATH = "META-INF/manifest.xml"
+
+NAMESPACE_URI = "urn:oasis:names:tc:opendocument:xmlns:manifest:1.0"
+MANIFEST_NAMESPACE = {"manifest": NAMESPACE_URI}
+PDF_MEDIA_TYPE = "application/pdf"
+
+ATTR_FULL_PATH = f"{{{NAMESPACE_URI}}}full-path"
+ATTR_MEDIA_TYPE = f"{{{NAMESPACE_URI}}}media-type"
+MANIFEST_FILE_ENTRY_TAG = "manifest:file-entry"
+
+
+def generate_text_checksum(text: str, algorithm: str = "sha256") -> str:
+    hash_func = getattr(hashlib, algorithm)()
+    hash_func.update(text.encode("utf-8"))
+    return hash_func.hexdigest()
+
+
+def generate_pdf_checksum(pdf_path: str, algorithm: str = "sha256") -> str:
+    text = extract_text(pdf_path)
+    return generate_text_checksum(text, algorithm)
+
+
+def get_pdf_path_in_adoc(adoc_archive: zipfile.ZipFile) -> str | None:
```

Comment:
The function either returns str or raises an exception, but never returns None. So str | None is misleading.